### ISSUE TITLE
zpl_random_range_f64 appears to generate garbage

### ISSUE THREAD

**Issue by cshenton** [zpl-c/zpl#67]
Seeing the following on windows 10 with `12.2.0`

```c
#define ZPL_IMPLEMENTATION
#include "zpl.h"

int main()
    zpl_random rand;
    zpl_random_init(&rand);
    for (int i=0; i < 100; i++)
    {
        zpl_printf("%f\n", zpl_random_range_f64(&rand, 0.0, 1.0));
    }
    return 0;
}
```

This program, which I'd expect to print random numbers between 0 and 1, randomly prints either `0.0`, `-0.0`, `2.0`, or `-2.0`.

Am I misunderstanding the API here or is this an implementation bug?

**Comment by cshenton** [comment]
For what it's worth, `(f32)zpl_random_gen_u32(&rand)  / (f32)ZPL_U32_MAX` seems to generate perfectly reasonable random numbers.

**Comment by zaklaus** [comment]
Thank you for the bug report, a fix has been deployed and is available as of `12.2.1`!

**Comment by cshenton** [comment]
Updated the `zpl.h` in my project to `12.2.1` and the fix is looking all good on my end.

Thanks for the quick fix and for the great library 😄
